### ISSUE TITLE
[FR] Document Public Members

### ISSUE THREAD

**Issue by Xazin** [AppFlowy-IO/appflowy-editor#298]
### Description

In an effort to improve our documentation, we should enforce comment docs on public members.

As we currently have too many non-documented public members, we shouldn't add in the rule until we have documented a majority of them.

Before we can add the rule, we should first add comment docs for public members in these directories:
- [ ] Comment Docs in `/lib/src/commands` [>90%]
- [ ] Comment Docs in `/lib/src/core` [>90%]
- [ ] Comment Docs in `/lib/src/editor` [>90%]
- [ ] Comment Docs in `/lib/src/extensions` [>90%]
- [ ] Comment Docs in `/lib/src/history` [>90%]
- [ ] Comment Docs in `/lib/src/infra` [>90%]
- [ ] Comment Docs in `/lib/src/plugins` [>90%]
- [ ] Comment Docs in `/lib/src/render` [>90%]
- [ ] Comment Docs in `/lib/src/service` [>90%]

_Note: It might be possible to circumvent the need for documentation of a member if it can be private. This might be something we will have to do to lessen the pain of this task, as well as clean up._

Each task itself should be opened in a separate PR, to cause the least amount of conflicts.

Once all the above has been completed, we can add the linter rule `public_member_api_docs`. Once there are <50 non-documented public members, we can change the analyzer to output an error instead of info.

### Impact

- Improved documentation on https://pub.dev
- Improved documentation inline whilst coding
- Improved documentation inline for users of `appflowy_editor`
